doc/user/sql/create-source/avro-source.md

```diff
@@ -0,0 +1,188 @@
+deletes existing data, or both. For more information, see [API Components:
+Envelopes](../../../overview/api-components#envelopes).
+
+Avro-encoded data sent through Kafka is the only type of data that can support
+the Debezium envelope.
+
+### Debezium envelope details
+
+The Debezium envelope provides a "diff envelope", which describes the decoded
+records' old and new values; this is roughly equivalent to the notion of Change
+Data Capture, or CDC. Materialize can use the data in this diff envelope to
+process data as representing inserts, updates, or deletes.
+
+This envelope is called the Debezium envelope because the only tool that we're
+aware of that includes this data is [Debezium].
+
+To use the Debezium envelope with Materialize, you must configure Debezium with
+your database.
+
+- [MySQL](https://debezium.io/documentation/reference/0.10/connectors/mysql.html)
+- [PostgreSQL](https://debezium.io/documentation/reference/0.10/connectors/postgresql.html)
+
+The Debezium envelope is only supported by sources published to Kafka by
+Debezium.
+
+#### Format implications
+
+Using the Debezium envelopes changes the schema of your Avro-encoded Kafka
+topics to include something akin to the following field:
+
+```json
+{
+    "type": "record",
+    "name": "envelope",
+    "fields": [
+        {
+        "name": "before",
+        "type": [
+            {
+            "name": "row",
+            "type": "record",
+            "fields": [
+                {"name": "a", "type": "long"},
+                {"name": "b", "type": "long"}
+            ]
+            },
+            "null"
+        ]
+        },
+        { "name": "after", "type": ["row", "null"] }
+    ]
+}
+```
+
+Note that:
+
+- If you use the Confluent Schema Registry to receive your schemas, you don't
+  need to manually create this field; Debezium will have taken care of it for
+  you.
+- The following section depends on the column's names and types, and is unlikely
+  to match our example:
+    ```json
+    ...
+    "fields": [
+            {"name": "a", "type": "long"},
+            {"name": "b", "type": "long"}
+        ]
+    ...
+    ```
+
+## Examples
+
+### Using a Confluent schema registry
+
+```sql
+CREATE SOURCE events
+FROM KAFKA BROKER 'localhost:9092' TOPIC 'events'
+FORMAT AVRO
+    USING CONFLUENT SCHEMA REGISTRY 'http://localhost:8081';
+```
+
+This creates a source that...
+
+- Automatically determines its schema from the Confluent Schema Registry.
+- Decodes data received from the `events` topic published by Kafka running on
+  `localhost:9092`.
+- Decodes using an Avro schema.
+- Is eligible to use the Debezium envelope because it's Avro-encoded and
+  published by Kafka; however, this still depends on whether or not the upstream
+  database publishes its data from a Debezium-enabled database.
+
+### Inlining the Avro schema
+
+```sql
+CREATE SOURCE user
+FROM KAFKA BROKER 'localhost:9092' TOPIC 'user'
+FORMAT AVRO
+USING SCHEMA '{
```

Comment:
This isn't a valid schema - did you copy it from somewhere and leave off part of it?

I think what happened is when you were deleting the inside of the `fields` array you accidentally deleted up to the wrong `]` delimiter.

Reply:
Ah, you're right. Going to simplify this down even further because the actual example is non-essential.